packages/app/src/components/popup/tx-item.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import type {
+  Transaction,
+  MempoolTransaction,
+  TransactionEventFungibleAsset,
+} from '@blockstack/stacks-blockchain-api-types';
+import { Box, Flex, Text } from '@stacks/ui';
+import { truncateMiddle, stacksValue } from '@common/stacks-utils';
+import BigNumber from 'bignumber.js';
+
+type Tx = MempoolTransaction | Transaction;
+
+const getAssetTransfer = (tx: Tx): TransactionEventFungibleAsset | undefined => {
+  if (tx.tx_type !== 'contract_call') return;
+  if (tx.tx_status !== 'success') return;
+  const transfer = tx.events.find(event => event.event_type === 'fungible_token_asset');
+  if (transfer?.event_type !== 'fungible_token_asset') return;
+  return transfer;
+};
+
+const getTxLabel = (tx: Tx): string => {
+  if (tx.tx_type === 'token_transfer') return 'Stacks Transfer';
+  if (tx.tx_status === 'pending') return 'Pending Transaction';
```

Comment:
if you have a pending tx token transfer it won't ever get here

Reply:
That is a very good catch